control-plane/gateways/deployment_dataplane_container.go

```diff
@@ -112,11 +109,10 @@ func consulDataplaneContainer(config GatewayConfig, containerConfig *v2beta1.Gat
 	wanPort := corev1.ContainerPort{
 		Name:          "wan",
 		ContainerPort: int32(constants.DefaultWANPort),
+		HostPort:      containerConfig.HostPort,
```

Comment:
Thinking out loud, should we potentially set a default value for `hostPort` if we happen to run into an issue where a host port isn't exposed? Super hypothetical

Reply:
I believe the defaultValue of host port as it exists today is 0, in which case the port won't be exposed. I could explicitly set that, but we get that behavior baked in when we initialize an empty struct.